core/CMakeLists.txt

```diff
@@ -77,6 +77,7 @@ set(SUBSURFACE_CORE_LIB_SRCS
 	# dirk ported some core functionality to c++.
 	qthelper.cpp
 	metadata.cpp
+	format.cpp
```

Comment:
do we need these functions on mobile as well? If yes, then https://github.com/Subsurface-divelog/subsurface/blob/master/packaging/ios/Subsurface-mobile/Subsurface-mobile.pro needs to be changed as well

Reply:
I don't think so?

They are used in
`core/profile.c`
`core/plannernotes.c`

But none of that seems to be used for mobile, since neither `RulerItem2` nor `ToolTipItem` is compiled in.

But since some of the `*_loc` functions still reside in `membuffer.c`, `format.c` will be needed for compilation.

Should I move `put_format_loc()` from `membuffer.c` to `format.c`?

Reply:
No, let's just add format.cpp to the Subsurface-mobile.pro file.
If all goes well we'll eventually be able to have a more interactive profile even on mobile, anyway...

Reply:
we don't support importing images from the mobile app, so that's not yet an issue...
(but of course, that's a feature that might be interesting to have there, given that more and more cameras allow connections to mobile devices to extract pictures)